recce/dbt.py

```diff
@@ -486,65 +486,90 @@ def refresh(self, refresh_file_path: str = None):
             elif refresh_file_path.endswith('catalog.json'):
                 self.base_catalog = load_catalog(refresh_file_path)
 
-    def compare_all_columns(self, primary_key: str, model: str):
-        # find the relation names
-        base_relation = self.generate_sql(f'{{{{ ref("{model}")  }}}}', True)
-
-        sql_template = f"""
-        {{{{
-            audit_helper.compare_all_columns(
-                a_relation=ref('{model}'),
-                b_relation='{base_relation}',
-                primary_key='{primary_key}'
-            )
-        }}}}
-        """
-        return self.execute_sql_lower_columns(sql_template, False)
-
-    def compare_relations(self, primary_key: str, model: str):
+    def get_base_relation(self, model):
         relation_holder = dict()
         with self.adapter.connection_named('test'):
             def callback(x):
                 relation_holder['relation'] = x
 
             tmp = f"""{{{{ config_base_relation(ref("{model}")) }}}}"""
             self.generate_sql(tmp, True, dict(config_base_relation=callback))
+        return relation_holder.get('relation')
+
+    def columns_value_mismatched_summary(self, primary_key: str, model: str):
+        column_results = []
+        primary_key_result = []
+
+        def log_callback(data, info=False):
+            if isinstance(data, tuple) and len(data) == 4 and 'perfect match' in data[1]:
+                column_results.append(data)
+
+                # full data for primary key to make added and removed
+                if data[0].lower() == primary_key.lower():
+                    primary_key_result.append(data)
```

Comment:
Case 1: If there are missing value, the value is wrong. 

![image](https://github.com/DataRecce/recce/assets/860633/7a38b7f3-cb64-41c1-8582-bf0c1d3b10dc)
![image](https://github.com/DataRecce/recce/assets/860633/1d8a0adb-8e51-4046-9477-f9d4e43348b9)


Case 2:
If both side are null, it should also mark as matched

Please checks all the 7 cases
```
            when a_query.EVENT_TYPE = b_query.EVENT_TYPE then '✅: perfect match'
            when a_query.EVENT_TYPE is null and b_query.EVENT_TYPE is null then '✅: both are null'
            when a_query.EVENT_ID is null then '🤷: ‍missing from a'
            when b_query.EVENT_ID is null then '🤷: missing from b'
            when a_query.EVENT_TYPE is null then '🤷: value is null in a only'
            when b_query.EVENT_TYPE is null then '🤷: value is null in b only'
            when a_query.EVENT_TYPE != b_query.EVENT_TYPE then '🙅: ‍values do not match'
            else 'unknown' -- this should never happen
```